app/grandchallenge/documentation/models.py

```diff
@@ -0,0 +1,136 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import models
+from django.db.models import Max
+from django.utils.translation import gettext_lazy as _
+from django_extensions.db.fields import AutoSlugField
+from simple_history.models import HistoricalRecords
+
+from grandchallenge.core.templatetags.bleach import clean
+from grandchallenge.core.utils.query import index
+from grandchallenge.subdomains.utils import reverse
+
+
+class LevelChoices(models.TextChoices):
+    """Documentation page level options"""
+
+    PRIMARY = "PRIMARY", _("Top level")
+    SECONDARY = "SECONDARY", _("Second level")
+    TERTIARY = "TERTIARY", _("Third level")
+
+
+class DocPageLevel:
+    """Documentation page level."""
+
+    DocPageLevelChoices = LevelChoices
+
+
+class DocPage(models.Model):
+
+    Level = DocPageLevel.DocPageLevelChoices
+
+    UP = "UP"
+    DOWN = "DOWN"
+    FIRST = "FIRST"
+    LAST = "LAST"
+
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    title = models.CharField(max_length=1024)
+    slug = AutoSlugField(populate_from="title", max_length=1024)
+    display_title = models.CharField(
+        max_length=255,
+        default="",
+        blank=True,
+        help_text=(
+            "On pages and in menu items, use this text. Spaces and special "
+            "chars allowed here. Optional field. If emtpy, title is used"
+        ),
+    )
+
+    content = models.TextField()
+
+    level = models.CharField(

```

Comment:
Are you sure this approach works for what you need? Would rather get the model right on the first migration.

Reply:
I've played around with a possible detail view already and we can definitely make it work with this model. For the detail view, I will need to add a few more property methods (apart from getting the next page, I will need a method to get a page's parent and its children), but that's not a problem. 

Are there any particular issues you are worried about?

Reply:
Or would it be better to add `parent` and `children` fields to the model rather than making these two things property methods?

Reply:
I added the parent and children methods together with the other changes so you get a better idea ;-)